### PR TITLE
Bugzilla 1423309 - Add the submission and user agent fields

### PR DESCRIPTION
**Pull request by trink**
r? @whd @fbertsch

### CODE REVIEW

File: moz_ingest/io_modules/decoders/moz_ingest/json.lua [hunk 1]
```diff
@@ -119,8 +131,18 @@ function transform_message(hsr, msg)
     process_json(hsr, msg)
 
     -- Migrate the original message data after the validation (avoids Field duplication in the error message)
-    msg.Hostname    = hsr:read_message("Hostname")
-    msg.Fields.Host = hsr:read_message("Fields[Host]")
+    msg.Hostname                = hsr:read_message("Hostname")
+    msg.Fields.Host             = hsr:read_message("Fields[Host]")
+    msg.Fields.http_user_agent  = hsr:read_message("Fields[User-Agent]")
```
Comment (trink) [inline moz_ingest/io_modules/decoders/moz_ingest/json.lua]: Should we keep the 'User-Agent' name here?  http_user_agent is what the web server logs use (and where the code was grabbed from)

Comment (whd) [reply moz_ingest/io_modules/decoders/moz_ingest/json.lua]: I don't have a preference.

**Comment by whd** [comment]
LGTM

File: moz_ingest/io_modules/decoders/moz_ingest/json.lua (hunk 1 above)
Comment (fbertsch) [reply moz_ingest/io_modules/decoders/moz_ingest/json.lua]: No need to keep it if `http_user_agent` is available as well.

Comment (trink) [reply moz_ingest/io_modules/decoders/moz_ingest/json.lua]: I renamed it for the pass through case (so I am not renaming headers)

**Pull request merged**